File: project/6.02_継承と変数/java/java_ex6_02_3/src/java_ex6_02_3/Ex6_02_3.java
package java_ex6_02_3;

class G {
	static int x = 100;
}

class H extends G {
	int x = 10;
	
	void display() {
		System.out.println( "Hクラスのx：" + x );
		System.out.println( "Gクラスのx：" + super.x );
		System.out.println( "Gクラスのx：" + G.x );
	}
}

public class Ex6_02_3 {

	public static void main(String[] args) {
		H h = new H();
		
		h.display();
		
	}

}

File: project/6.02_継承と変数/py/py_ex6_02_3/src/ex6_02_3.py
'''
Created on 2023/05/07

@author: Keroichi.T
'''

class G( object ):
    x = 100
    
    def __init__(self):
        self.y = 200
    
class H( G ):
    x = 10
    
    def __init__(self):
        super().__init__()
        self.y = 20
    
    def display(self):
        print( f"Hクラスのx:{H.x}" )
        print( f"Gクラスのx:{super().x}" )
        print( f"Gクラスのx:{G.x}" )

def main():
    h = H()
    
    h.display()

if __name__ == '__main__':
    main()